pkg/handler/v1/common/common.go

```diff
@@ -75,8 +75,9 @@ type OIDCConfiguration struct {
 // UpdateManager specifies a set of methods to handle cluster versions & updates
 type UpdateManager interface {
 	GetVersions(string) ([]*version.Version, error)
+	GetVersionsV2(string, kubermaticv1.ProviderType, ...version.IncompatibilityCondition) ([]*version.Version, error)
```

Comment:
Why did you decide to do it this way? Why not simply extend the existing `GetVersions` function? You were able to extend the `GetPossibleUpdates` already.

Reply:
I did in this way because GetVersions is consumed by [this](https://github.com/kubermatic/kubermatic/blob/489c811eebf2419af4e83c8b1607fcd5a360daba/pkg/handler/routes_v1.go#L1668-L1689) function, which has no way to fetch provider information, therefore must be replaced. I think it shouldn't be deleted before the UI has been migrated to use the new endpoint instead ([this](https://github.com/kubermatic/kubermatic/blob/1cd656ec7264744f3752b832273a43cf497ab211/pkg/handler/v2/routes_v2.go#L3554-L3579) one). Once The new endpoint will be correctly used by the UI, I think we can get rid of the old endpoint and change the getVersions function signature as you suggested.

Reply:
I see. I am so scared that the two versions will live forever and we will never clean up... so scared.. can you calm me down by promising you will clean it up? ^^

Reply:
Once the UI part will be implemented I'll clean it up, you have my word :grin: